### PR TITLE
feat: support for branches and content-types pagination. 

### PR DESCRIPTION
**Pull request by vkalta**
This PR resolves CS-38229.

### CODE REVIEW

File: src/commands/tsgen.ts [hunk 1]
```diff
@@ -45,6 +45,14 @@ export default class TypeScriptCodeGeneratorCommand extends Command {
       default: true,
       allowNo: true,
     }),
+
+    branch: flags.string({
+      char: 'b',
```
Comment (netrajpatel) [inline src/commands/tsgen.ts:50]: @vkalta Remove this short flag, current cli does not have any short flag for a branch flag.

File: src/commands/tsgen.ts [hunk 2]
```diff
@@ -45,6 +45,14 @@ export default class TypeScriptCodeGeneratorCommand extends Command {
...
+      description: 'branch',
+      hidden: false,
+      multiple: false,
+      required: true,
```
Comment (netrajpatel) [inline src/commands/tsgen.ts:54]: @vkalta Can you make this branch flag optional? We have some customers who do not have branches enabled yet.

File: src/lib/stack/client.ts [hunk 3]
```diff
@@ -16,22 +18,56 @@ export type StackConnectionConfig = {
...
-    )
-
-    const results = await stack.getContentTypes({
-      include_global_field_schema: true,
+    const stack = client({
+      api_key: config.apiKey,
+      delivery_token: config.token,
+      environment: config.environment,
+      region: config.region,
+      branch: config.branch,
```
Comment (netrajpatel) [inline src/lib/stack/client.ts:39]: @vkalta This branch param should be optional

**Review by netrajpatel — CHANGES_REQUESTED** [review]

File: src/commands/tsgen.ts (hunk 1 above)
Comment (netrajpatel) [reply src/commands/tsgen.ts]: @vkalta Can you please check this?

**Comment by aman19K** [comment]
Hey @vkalta , Can you pls update the version ? version": "2.1.2 already released.
https://github.com/Contentstack-Solutions/contentstack-cli-tsgen/blob/master/package.json

cc: @netrajpatel

**Review by aravindbuilt — APPROVED** [review]

**Review by netrajpatel — APPROVED** [review]

**Pull request merged**